federation/README.md

```diff
@@ -0,0 +1,318 @@
+    .
+    .
+
+    federation {
+        bundle_endpoint {
+            address = "0.0.0.0"
+            port = 8443
+        }
+    }
+}
+```
+This will publish the federation endpoint in any IP address at port 8443 in the host where the SPIRE Server is running.
+
+On the other side, the stocks market service provider's SPIRE Server is configured in a similar fashion:
+```hcl
+server {
+    .
+    .
+    trust_domain = "stocksmarket.org"
+    .
+    .
+
+    federation {
+        bundle_endpoint {
+            address = "0.0.0.0"
+            port = 8443
+        }
+    }
+}
+```
+
+At this point, both SPIRE Servers expose their federation endpoints to provide their trust bundles, but none of them knows how to reach each other's federation endpoint. 
+
+## Using WebPKI authentication
+
+We are going to assume that only the broker's SPIRE Server will use WebPKI authentication for its federation endpoint, the stocks market SPIRE Server will still use SPIFFE authentication. Hence, the stocks market SPIRE Server configuration remains the same as seen in the previous section.
+
+Then, to configure the broker's SPIRE Server bundle endpoint, we configure the `federation` section as follows:
+
+```hcl
+server {
+    .
+    .
+    trust_domain = "broker.org"
+    .
+    .
+
+    federation {
+        bundle_endpoint {
+            address = "0.0.0.0"
+            port = 443
+            acme {
+                domain_name = "broker.org"
+                email = "some@email.com"
+                tos_accepted = true
+            }   
+        }
+    }
+}
+```
+This will publish the federation endpoint in any IP address at port 443. We use port 443 because we will be using Let's Encrypt as our ACME provider (this is used by default, if you want to use a different one then you must set the `directory_url` configurable). Note that `tos_accepted` was set to `true`, meaning that we accept the terms of service of our ACME provider, which in turn is needed when using Let's Encrypt.
+
+Worth noting that for this to work you must own the `broker.org` domain and it must point to the SPIRE Server exposing the federation bundle endpoint.
+
+# Configure the SPIRE Servers to retrieve trust bundles from each other
+
+This is the other part of the SPIRE Server configuration needed in order to achieve federation, we need to tell each SPIRE Server where it can find the trust bundles for other trust domains, in other words, we have to configure the `federate_with` section. The configuration of this section has some slight differences when using the different methods of authentication.
+
+## Using SPIFFE authentication
+
+As we saw previously, the SPIRE Server of the stocks market service provider has its federation endpoint listening on port `8443` at any IP. We will also assume that `spire-server-stocks` is a DNS name that resolves to the stocks market service's SPIRE Server IP. Then, the broker's SPIRE Server must be configured with the following `federate_with` section:
```

Comment:
No, it is not, because `stockmarket.org` is a SPIFFE trust domain, not a domain name. The actual host has the `spire-server-stocks` name as configured in the docker-compose.yml file.

Reply:
OK, I understand. Please do use the phrase "IP address" though, rather than just IP.

Reply:
Oh yeah, I fixed that. Thanks